src/tests/PostTest.test.ts

```diff
@@ -290,29 +291,126 @@ describe('post tests', () => {
     const userController = ControllerFactory.user(conn);
     const post = PostFactory.fakeTemplate();
     post.user = UserFactory.fakeTemplate();
+    post.user.saved = [];
 
     await new DataFactory()
       .createPosts(post)
       .createUsers(post.user)
       .write();
 
-    const postAndUserUuidParam = new PostAndUserUuidParam();
-    postAndUserUuidParam.postId = '81e6896c-a549-41bf-8851-604e7fbd4f1f';
-    postAndUserUuidParam.userId = '81e6896c-a549-41bf-8851-604e7fbd4f1f';
-
     let userResponse = (await userController.getUserByGoogleId('shungoGoogleID')).user;
     expect(userResponse).not.toBeUndefined();
     expect(userResponse?.saved).toEqual([]);
 
-    let getPostResponse = await postController.savePost(postAndUserUuidParam);
+    await postController.savePost(post.user, uuidParam);
     userResponse = (await userController.getUserByGoogleId('shungoGoogleID')).user;
     if (userResponse) {
       userResponse.saved[0].price = Number(userResponse.saved[0].price);
     }
     expect(userResponse?.saved).toEqual([expectedPost]);
 
-    getPostResponse = await postController.unsavePost(postAndUserUuidParam);
+    await postController.unsavePost(post.user, uuidParam);
     userResponse = (await userController.getUserByGoogleId('shungoGoogleID')).user;
     expect(userResponse?.saved).toEqual([]);
   });
+
+  test('get all archived posts - no posts', async () => {
+    const postController = ControllerFactory.post(conn);
+
+    const getPostsResponse = await postController.getArchivedPosts();
+
+    expect(getPostsResponse.posts).toHaveLength(0);
+  });
+
+  test('get all archived posts - one post', async () => {
+    const postController = ControllerFactory.post(conn);
+    const post = PostFactory.fake();
+    post.user = UserFactory.fake();
+    post.archive = true;
+
+    await new DataFactory()
+      .createPosts(post)
+      .createUsers(post.user)
+      .write();
+
+    const getPostsResponse = await postController.getArchivedPosts();
+
```

Comment:
minimize the newlines in a test to 2 since we want to have 3 chunks in our code (Arrange-Act-Assert https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/)